### PR TITLE
Document GSoC project selection and ranking process

### PR DESCRIPTION
**Pull request by aliok**
Document GSoC project selection and ranking process

Based on the discussion here: https://github.com/cncf/mentoring/discussions/860

### CODE REVIEW

**Comment by aliok** [comment]
One note: we want to encourage folks who are from diverse backgrounds and who are currently underrepresented in the tech industry.
1. I didn't see anything around this in Google's [program rules](https://summerofcode.withgoogle.com/rules). I was looking for some material to reference to.
2. Documenting it part of the selection and ranking process might need a legal review from CNCF.
3. Need to check CNCF materials around diversity
4. Need to check other organizations participating in GSoC for some references.

I think, documenting the process around this matter should be a separate ticket.

**Review by nate-double-u — APPROVED** [review]
Hi @aliok, thanks for putting this together. This is as I recall our discussion going.

I'm approving, but think we should leave this PR open till **May 2nd** for community comment.

File: mentoring-wg/gsoc-selection-and-ranking-process.md
Comment (aliok) [inline mentoring-wg/gsoc-selection-and-ranking-process.md:19]: We should add something around the applicant profile, as in @tanayvaswani 's feedback

Comment (nate-double-u) [reply mentoring-wg/gsoc-selection-and-ranking-process.md:19]: Is there a profile as such to review? I know a candidate's name is provided, but other than that it's only the proposal that is shared in the system (IIRC).

Comment (aliok) [reply mentoring-wg/gsoc-selection-and-ranking-process.md:19]: The profile is often shared in the proposal. Who the applicant is, what kind of contributions they've made to the project already, or to other projects. I will just add a single sentence about this in the criteria.

File: mentoring-wg/gsoc-selection-and-ranking-process.md
Comment (aliok) [inline mentoring-wg/gsoc-selection-and-ranking-process.md:17]:
```suggestion
    - Proposal quality (lower weight, as GSoC admins might not know the technical details of the project)
    - Applicant profile that shows the applicant is capable of implementing the project, such as contributions to other projects
```

**Comment by aliok** [comment]
I will wait for another week on this PR for any additional feedback and then merge.

**Comment by aliok** [comment]
Merging!

**Pull request merged**
